community/kernel-api/src/main/java/org/neo4j/internal/kernel/api/SchemaWrite.java

```diff
@@ -47,18 +57,42 @@
     void indexDrop( IndexReference index ) throws SchemaKernelException;
 
     /**
-     * Create unique property constraint
+     * Create unique property constraint. Default configured index provider will be used.
      *
      * @param descriptor description of the constraint
      */
-    ConstraintDescriptor uniquePropertyConstraintCreate( SchemaDescriptor descriptor ) throws SchemaKernelException;
+    default ConstraintDescriptor uniquePropertyConstraintCreate( SchemaDescriptor descriptor ) throws SchemaKernelException
+    {
+        return uniquePropertyConstraintCreate( descriptor, null );
+    }
 
     /**
-     * Create node key constraint
+     * Create unique property constraint. Optionally a specific provider name can be specified.
      *
      * @param descriptor description of the constraint
+     * @param providerName specific index provider this index will be created for. If {@code null} then the default configured will be used.
+     * @return the newly created index
+     */
+    ConstraintDescriptor uniquePropertyConstraintCreate( SchemaDescriptor descriptor, String providerName ) throws SchemaKernelException;
```

Comment:
`@Nullable`

Reply:
It's not available as a dependency in this component, I don't wanna mess with that.

Reply:
Agree, that's fine then since javadoc is super clear!